tests/testsuite/init.rs

```diff
@@ -598,3 +598,68 @@ mod tests {
 "#
     );
 }
+
+#[cargo_test]
+fn creates_binary_when_instructed_and_has_lib_file_no_warning() {
+    let path = paths::root().join("foo");
+    fs::create_dir(&path).unwrap();
+    fs::write(path.join("foo.rs"), "fn not_main() {}").unwrap();
+    cargo_process("init --bin")
+        .cwd(&path)
+        .with_stderr(
+            "[WARNING] file 'foo.rs' seems to be a library file\n[CREATED] binary (application) package"
```

Comment:
I'm not sure in this case we should issue a warning. It means there's no `fn main` most likely

Reply:
The warning seems fine to me.  I imagine the user can figure things out afterwards.